### PR TITLE
Refactor memory charts.

### PR DESCRIPTION
**Pull request by polina-c**
No functional changes.
Move functionality from widgets to controllers and unify widgets for chart types into one widget class.

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/panes/chart/controller/charts/android_chart_controller.dart [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+import '../../../../../../shared/charts/chart_controller.dart';
+import '../../../../../../shared/charts/chart_trace.dart' as chart_trace;
+import '../../../../../../shared/charts/chart_trace.dart'
+    show ChartType, ChartSymbol;
+import '../../../../shared/primitives/memory_timeline.dart';
+import '../../data/charts.dart';
+
+// ignore: avoid_classes_with_only_static_members, enum-like classes are ok
+class _Color {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/chart/controller/charts/android_chart_controller.dart:17]: use advanced enum instead

File: packages/devtools_app/lib/src/screens/memory/panes/chart/controller/charts/vm_chart_controller.dart [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+import '../../../../../../shared/charts/chart_controller.dart';
+import '../../../../../../shared/charts/chart_trace.dart' as chart_trace;
+import '../../../../../../shared/charts/chart_trace.dart'
+    show ChartType, ChartSymbol;
+import '../../../../shared/primitives/memory_timeline.dart';
+import '../../data/charts.dart';
+
+// ignore: avoid_classes_with_only_static_members, enum-like classes are ok
+class _Colors {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/panes/chart/controller/charts/vm_chart_controller.dart:17]: nut use advanced enum

File: packages/devtools_app/lib/src/screens/memory/panes/chart/widgets/legend.dart [hunk 3]
```diff
@@ -197,30 +196,42 @@ class LegendRow extends StatelessWidget {
...
+class _Names {
+  static const manualSnapshot = 'Snapshot';
+  static const autoSnapshot = 'Auto';
+  static const monitor = 'Monitor';
+  static const reset = 'Reset';
+  static const vmGC = 'GC VM';
+  static const manualGC = 'Manual';
+  static const event = 'Event';
+  static const events = 'Events';
+}
```
[the remark this answers is not in the document]
Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/screens/memory/panes/chart/widgets/legend.dart:209]: Also, this seems to overlap with the `EventsTraceName` class. Do we need to duplicate the set of names / categories?

**Review by bkonyi — COMMENTED** [review]
I had the same comments as Kenzie. Otherwise, LGTM.

File: packages/devtools_app/lib/src/screens/memory/panes/chart/controller/charts/android_chart_controller.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/chart/controller/charts/android_chart_controller.dart:17]: Some colors are not consts, so advanced enums are not applicable :(

File: packages/devtools_app/lib/src/screens/memory/panes/chart/widgets/legend.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/panes/chart/widgets/legend.dart]: Added TODO.

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**